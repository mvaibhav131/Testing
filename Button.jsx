

export const Button=({variant,colorScheme,children,onClick})=>{
    return (
    <button 
    data-testid="customButton"
    onClick={onClick}
    className={`btn ${colorScheme || "red"} ${variant || "ghost"}`}>{children}</button>
    );
};