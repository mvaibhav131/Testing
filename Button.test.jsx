
import App from "../App";
import { fireEvent, render,screen,} from "@testing-library/react";
import { Button } from "./Button";
import renderer from "react-test-renderer";

describe("Button Tests", function(){
  
    it("should have button in dom",function(){
        render(<Button>Click me</Button>);
        let button = screen.getByText("Click me");
        expect(button).toBeInTheDocument();
    });
    it("should render based on testid", () =>{
       render(<Button>Click me</Button>); //you also render the App because app file and inside app Button is render.
       let button=screen.getByTestId("customButton");
       expect(button).toBeInTheDocument();
    });
     it("should change theme",function(){
        render(<App></App>);
        let h3 =screen.getByText("The theme is light");
        expect(h3).toHaveTextContent("light");

        let themeButton=screen.getByText("Click me");
        fireEvent.click(themeButton);
        expect(h3).toHaveTextContent("dark");
    });

    it("should called given function",function() {
        const mockfn=jest.fn();
        render(<Button onClick={mockfn}>Click me</Button>);
        const btn=screen.getByText("Click me");
        fireEvent.click(btn);
        expect(mockfn).toBeCalled();
    });
    //first install the npm i react-test-renderer --save-dev
    //is create the new snapshot name file inside component of that button show it in json format 
    it("Should match snapshot",function(){
        const tree = renderer.create(  <Button colorScheme={"green"} variant="bordered">Add Counter</Button>)
         // .toJSON();   
        expect(tree).toMatchSnapshot();
    });
});