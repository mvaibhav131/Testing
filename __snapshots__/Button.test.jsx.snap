// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Should match snapshot 1`] = `
<button
  className="btn green bordered"
  data-testid="customButton"
>
  Add Counter
</button>
`;
